internal/controller/drplacementcontrol_watcher.go

```diff
@@ -245,7 +255,17 @@ func DRPolicyPredicateFunc() predicate.Funcs {
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			log.Info("Update event")
 
-			return RequiresDRPCReconciliation(e.ObjectOld.(*rmn.DRPolicy), e.ObjectNew.(*rmn.DRPolicy))
+			drpOld, ok := e.ObjectOld.(*rmn.DRPolicy)
+			if !ok {
+				return false
+			}
+
+			drpNew, ok := e.ObjectNew.(*rmn.DRPolicy)
+			if !ok {
+				return false
+			}
+
+			return RequiresDRPCReconciliation(drpOld, drpNew)
```

Comment:
We repeat the same boilerplate 3 times, maybe RequiresDRPCReconciliation should do these checks instead of the callers?

Reply:
The boiler plate is not the same each deals with different struct, DRPC, DRPolicy, DRCluster.